runsc/container/container.go

```diff
@@ -909,11 +944,16 @@ func (c *Container) forEachSelfMount(fn func(mountSrc string)) {
 // createGoferFilestores creates the regular files that will back the
 // tmpfs/overlayfs mounts that will overlay some gofer mounts. It also returns
 // information about how each gofer mount is configured.
-func (c *Container) createGoferFilestores(ovlConf config.Overlay2, mountHints *boot.PodMountHints) ([]*os.File, []boot.GoferMountConf, error) {
+func (c *Container) createGoferFilestores(ovlConf config.Overlay2, mountHints *boot.PodMountHints, rootfsHint *boot.RootfsHint) ([]*os.File, []boot.GoferMountConf, error) {
 	var goferFilestores []*os.File
 	var goferConfs []boot.GoferMountConf
 
 	// Handle root mount first.
+	if rootfsHint != nil && rootfsHint.Overlay != "" {
+		if err := ovlConf.Set(fmt.Sprintf("root:%s", rootfsHint.Overlay)); err != nil {
+			return nil, nil, fmt.Errorf("failed to override rootfs overlay config via rootfs hint: %v", err)
+		}
```

Comment:
> this will overwrite configurations like `all:*` that would have otherwise applied to the submounts.

Oh, my bad, this was supposed to be a copy of `ovlConf`, and we just need to modify the copy and pass the copy instead of `ovlConf` to `c.createGoferFilestore()` when creating the gofer filestore for rootfs.

> Instead, we should pass `OverlayMedium` type to `createGoferFilestore()` function. See previous comment about this type.

Good idea! Will do. Thanks for the suggestions!